ddsrouter_event/include/ddsrouter_event/wait/CounterWaitHandler.hpp

```diff
@@ -0,0 +1,132 @@
+ *
+ * This works as an int value that allows to wait for a specific value or threshold.
+ */
+class CounterWaitHandler : protected WaitHandler<CounterType>
+{
+public:
+
+    /**
+     * @brief Construct a new Boolean Wait Handler object
+     *
+     * @param value to initialize internal value
+     * @param enabled whether the object starts enabled or disabled
+     */
+    CounterWaitHandler(
+            CounterType value,
+            bool enabled = true);
+
+    //! Default constructor
+    ~CounterWaitHandler();
+
+    /////
+    // Enabling methods
+
+    // Make this methods public
+    using WaitHandler<CounterType>::enable;
+    using WaitHandler<CounterType>::disable;
+    using WaitHandler<CounterType>::enabled;
+    using WaitHandler<CounterType>::set_value;
+    using WaitHandler<CounterType>::get_value;
+    using WaitHandler<CounterType>::stop_and_continue;
+
+    /////
+    // Wait methods
+
+    /**
+     * @brief Wait current thread while value is not \c expected_value
+     *
+     * This thread will be awake when counter is equal to \c expected_value .
+     *
+     * @param expected_value value of counter which the thread will awake
+     * @param timeout maximum time in milliseconds that should wait until awaking for timeout
+     *
+     * @return reason why thread was awake
+     */
+    AwakeReason wait_equal(
+            CounterType expected_value,
+            const utils::Duration_ms& timeout = 0);
+
+    /**
+     * @brief Wait current thread while internal value is lower or equal than \c upper_bound .
+     *
+     * @param upper_bound maximum value of counter which with the thread will not awake
+     * @param timeout maximum time in milliseconds that should wait until awaking for timeout
+     *
+     * @return reason why thread was awake
+     */
+    AwakeReason wait_greater_than(
+            CounterType upper_bound,
+            const utils::Duration_ms& timeout = 0);
+
+    //! @brief Wait current thread while internal value is lower than \c upper_bound .
+    AwakeReason wait_greater_equal_than(
+            CounterType upper_bound,
+            const utils::Duration_ms& timeout = 0);
+
+    /**
+     * @brief Wait current thread while internal value is higher or equal than \c lower_bound .
+     *
+     * @param lower_bound minimum value of counter which with the thread will not awake
+     * @param timeout maximum time in milliseconds that should wait until awaking for timeout
+     *
+     * @return reason why thread was awake
+     */
+    AwakeReason wait_lower_than(
+            CounterType lower_bound,
+            const utils::Duration_ms& timeout = 0);
+
+    //! Wait current thread while internal value is higher than \c lower_bound .
+    AwakeReason wait_lower_equal_than(
+            CounterType lower_bound,
+            const utils::Duration_ms& timeout = 0);
+
+    /////
+    // Value methods
+
+    //! Operator ++ to add 1 to internal value
+    CounterWaitHandler& operator++();
```

Comment:
NIT: perhaps also add the postincrement operator.

Reply:
I think it would be ambiguous. When using `waiter++` what should it return? In theory should be a new object which counter is +1 than the object that has been called. But we do not want that, but to add 1 to the object being called.
Using prefix will guarantee that the object will be incremented AND that the result is that same object, which is the expected behaviour.